### ISSUE TITLE
Apache 2.4.26 + php-fpm - prepareBaseUrl() not working

### ISSUE THREAD

**Issue by lukfir** [symfony/symfony#23358]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4

After update Apache to 2.4.26 version is a problem with `function prepareBaseUrl()` in `Symfony\Component\HttpFoundation\Request` class when you also are using fcgi.

The problem exist because in `$_SERVER['SCRIPT_NAME']` and `$_SERVER['PHP_SELF']` variables are storing non-expected values when you are using default Apache configuration. In these variables do not have directories is only file name.

For example these are expected values:
```
_SERVER["QUERY_STRING"] controller=AdminDashboard&token=xxx
_SERVER["REQUEST_URI"] /presta17/admin408pewuj9/index.php?controller=AdminDashboard&token=xxx
_SERVER["SCRIPT_NAME"] /presta17/admin408pewuj9/index.php
_SERVER["PHP_SELF"] /presta17/admin408pewuj9/index.php
```
But these are real results:
```
_SERVER["QUERY_STRING"] controller=AdminDashboard&token=xxx
_SERVER["REQUEST_URI"] /presta17/admin408pewuj9/index.php?controller=AdminDashboard&token=xxx
_SERVER["SCRIPT_NAME"] /index.php
_SERVER["PHP_SELF"] /index.php
```

The problem exist in the newest Apache version [http://www.apache.org/dist/httpd/CHANGES_2.4.26](http://www.apache.org/dist/httpd/CHANGES_2.4.26) because Apache developers added varibles `ProxyFCGISetEnvIf` and `ProxyFCGIBackendType`. 

The fastest way to fix the problem is set `ProxyFCGIBackendType GENERIC` in Apache configuration (also in .htaccess), but I am not sure that this is the best solution.

So, You should require additional configuration for described environments or change base url preparing method.

**Comment by mbrodala** [comment]
I can confirm the basic issue itself (noticed in TYPO3), even outputting `$_SERVER` shows that `SCRIPT_NAME` and `PHP_SELF` are broken with Apache 2.4.26. The suggested fix indeed brings back the full file path.

**Comment by mbrodala** [comment]
Also see the related upstream report:

https://bz.apache.org/bugzilla/show_bug.cgi?id=61202

**Comment by jakzal** [comment]
The bug has been fixed in [Apache 2.4.27](https://svn.apache.org/viewvc?view=revision&revision=1800618) (change has been reverted).
